backend/LeetWars.Core/LeetWars.Core.BLL/Services/UserService.cs

```diff
@@ -181,16 +192,106 @@ public async Task<UserFullDto> UpdateUserRankAsync(EditUserDto userDto)
         return _mapper.Map<UserFullDto>(user);
     }
 
-    public async Task<List<UserDto>> GetLeaderBoardAsync(PageSettingsDto? page)
+    public async Task<List<UserDto>> GetLeaderBoardAsync(LeaderBoardPageSettingsDto page)
     {
-        var users = _context.Users.OrderByDescending(u => u.TotalScore).AsQueryable();
-        if (page is not null)
+        var query = _context.Users.OrderByDescending(u => u.TotalScore).AsQueryable();
+
+        if (page.GetFriends)
+        {
+            var currentUser = await GetCurrentUserEntityAsync();
+            var currentUserDto = _mapper.Map<UserDto>(currentUser);
+
+            var currentUserFriendsIds = currentUserDto.Friendships?
+                .Where(f => f.FriendshipStatus == FriendshipStatus.Accepted)
+                .Select(f => f.FriendId) ?? throw new NotFoundException(nameof(List<User>));
+
+            query = query.Where(u => currentUserFriendsIds.Contains(u.Id));
+        }
+
+        query = query.Skip(page.PageSize * (page.PageNumber - 1)).Take(page.PageSize);
+
+        return _mapper.Map<List<UserDto>>(await query.ToListAsync());
+    }
+
+    public async Task<UserDto> SendFriendshipRequestAsync(NewFriendshipDto newFriendshipDto)
+    {
+        var senderId = newFriendshipDto.SenderId;
+        var recipientId = newFriendshipDto.RecipientId;
+        var currentUser = await GetCurrentUserEntityAsync();

```

Comment:
the same here, it would be better to load the user here, do not use GetCurrentUserEntityAsync, because it consists a lot of unnecessary data